src/test/mempool_tests.cpp

```diff
@@ -101,4 +102,56 @@ BOOST_AUTO_TEST_CASE(MempoolRemoveTest)
     removed.clear();
 }
 
+BOOST_AUTO_TEST_CASE(RemoveWithoutBranchId) {
+    CTxMemPool pool(CFeeRate(0));
+
+    // Add some Sprout transactions
+    for (auto i = 1; i < 11; i++) {
+        CMutableTransaction tx = CMutableTransaction();
+        tx.vout.resize(1);
+        tx.vout[0].scriptPubKey = CScript() << OP_11 << OP_EQUAL;
+        tx.vout[0].nValue = i * COIN;
+        pool.addUnchecked(tx.GetHash(), CTxMemPoolEntry(tx, 0, 0, 0.0, 1, true, NetworkUpgradeInfo[Consensus::BASE_SPROUT].nBranchId));
+    }
+    BOOST_CHECK_EQUAL(pool.size(), 10);
+
+    // Check the pool only contains Sprout transactions
+    for (std::map<uint256, CTxMemPoolEntry>::const_iterator it = pool.mapTx.begin(); it != pool.mapTx.end(); it++) {
+        BOOST_CHECK_EQUAL(it->second.GetBranchId(), NetworkUpgradeInfo[Consensus::BASE_SPROUT].nBranchId);
+    }
+
+    // Add some dummy transactions
+    for (auto i = 1; i < 11; i++) {
+        CMutableTransaction tx = CMutableTransaction();
+        tx.vout.resize(1);
+        tx.vout[0].scriptPubKey = CScript() << OP_11 << OP_EQUAL;
+        tx.vout[0].nValue = i * COIN + 100;
+        pool.addUnchecked(tx.GetHash(), CTxMemPoolEntry(tx, 0, 0, 0.0, 1, true, NetworkUpgradeInfo[Consensus::UPGRADE_TESTDUMMY].nBranchId));
+    }
+    BOOST_CHECK_EQUAL(pool.size(), 20);
+
+    // Add some Overwinter transactions
+    for (auto i = 1; i < 11; i++) {
+        CMutableTransaction tx = CMutableTransaction();
```

Comment:
This isn't actually an Overwinter transaction.  Need to set the overwintered flag, version and version group id.

Reply:
Yep, which requires rebasing on the transaction format PR. Will do.